src/object.ts

```diff
@@ -0,0 +1,538 @@
+        }
+        return parts.join('/').toLowerCase();
+    }
+
+    /** Return root of API path up to API version. */
+    protected apiVersionPath(apiVersion: string): string {
+        const api = apiVersion.includes('/') ? 'apis' : 'api';
+        return [this.basePath, api, apiVersion].join('/');
+    }
+
+    /**
+     * Merge default headers and provided headers, setting the 'Accept' header to 'application/json' and, if the
+     * `action` is 'PATCH', the 'Content-Type' header to [[KubernetesPatchStrategies.StrategicMergePatch]].  Both of
+     * these defaults can be overriden by values provided in `optionsHeaders`.
+     *
+     * @param optionHeaders Headers from method's options argument.
+     * @param action HTTP action headers are being generated for.
+     * @return Headers to use in request.
+     */
+    protected generateHeaders(
+        optionsHeaders: { [name: string]: string },
+        action: string = 'GET',
+    ): { [name: string]: string } {
+        const headers: { [name: string]: string } = Object.assign({}, this._defaultHeaders);
+        headers.accept = 'application/json';
+        if (action === 'PATCH') {
+            headers['content-type'] = KubernetesPatchStrategies.StrategicMergePatch;
+        }
+        Object.assign(headers, optionsHeaders);
+        return headers;
+    }
+
+    /**
+     * Get metadata from Kubernetes API for resources described by `kind` and `apiVersion`.  If it is unable to find the
+     * resource `kind` under the provided `apiVersion`, `undefined` is returned.
+     *
+     * @param apiVersion Kubernetes API version, e.g., 'v1' or 'apps/v1'.
+     * @param kind Kubernetes resource kind, e.g., 'Pod' or 'Namespace'.
+     * @return Promise of the resource metadata or `undefined` if the resource is not found.
+     */
+    protected async resource(apiVersion: string, kind: string): Promise<V1APIResource | undefined> {
+        // verify required parameter 'apiVersion' is not null or undefined
+        if (apiVersion === null || apiVersion === undefined) {
+            throw new Error('Required parameter apiVersion was null or undefined when calling resource');
+        }
+        // verify required parameter 'kind' is not null or undefined
+        if (kind === null || kind === undefined) {
+            throw new Error('Required parameter kind was null or undefined when calling resource');
+        }
+
+        const localVarPath = this.apiVersionPath(apiVersion);
+        const localVarQueryParameters: any = {};
+        const localVarHeaderParams = this.generateHeaders({});
+
+        const localVarRequestOptions: request.Options = {
+            method: 'GET',
+            qs: localVarQueryParameters,
+            headers: localVarHeaderParams,
+            uri: localVarPath,
+            useQuerystring: this._useQuerystring,
+            json: true,
+        };
+
+        try {
+            const getApiResponse = await this.requestPromise<V1APIResourceList>(
```

Comment:
Probably want to cache this lookup in memory in the client for the sake of performance.

Not required to merge this PR, but at least add a TODO and file an issue.

(or just add it to this PR)

Reply:
How long should it be cached? The response will change as CRDs come and go. Does the client already have a caching solution that can be used here?

Reply:
I would cache until you fail to find something, and then I would try reloading.

Reply:
`kubectl` caches in `~/.kube/schema` fwiw.

Reply:
Cache added.